gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,800 @@
+    this.httpErrorMessage = builder.httpErrorMessage;
+    this.description = builder.description;
+  }
+
+  /**
+   * Returns the operation's {@code Compute} object used to issue requests.
+   */
+  public Compute compute() {
+    return compute;
+  }
+
+  /**
+   * Returns the service-defined unique identifier for the operation.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the creation timestamp in milliseconds since epoch.
+   */
+  public Long creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns the operation's identity. This method returns an {@link GlobalOperationId} for global
+   * operations, returns a {@link RegionOperationId} for region operations and returns a
+   * {@link ZoneOperationId} for zone operations.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  @SuppressWarnings("unchecked")
+  public <T extends OperationId> T operationId() {
+    return (T) operationId;
+  }
+
+  /**
+   * Reserved for future use.
+   */
+  String clientOperationId() {
+    return clientOperationId;
+  }
+
+  /**
+   * Returns the type of operation.
+   */
+  public String operationType() {
+    return operationType;
+  }
+
+  /**
+   * Returns the URL of the resource that the operation is modifying.
+   */
+  public String targetLink() {
+    return targetLink;
+  }
+
+  /**
+   * Returns the unique service-defined target ID, which identifies the resource that the operation
+   * is modifying.
+   */
+  public String targetId() {
+    return targetId;
+  }
+
+  /**
+   * Returns the status of the operation.
+   */
+  public Status status() {
+    return status;
+  }
+
+  /**
+   * Returns an optional textual description of the current status of the operation.
+   */
+  public String statusMessage() {
+    return statusMessage;
+  }
+
+  /**
+   * Returns the user who requested the operation, for example: {@code user@example.com}.
+   */
+  public String user() {
+    return user;
+  }
+
+  /**
+   * Returns an optional progress indicator that ranges from 0 to 100. There is no requirement that
+   * this be linear or support any granularity of operations. This should not be used to guess when
+   * the operation will be complete. This number should monotonically increase as the operation
+   * progresses.
+   */
+  public Integer progress() {
+    return progress;
+  }
+
+  /**
+   * Returns the time that this operation was requested. In milliseconds since epoch.
+   */
+  public Long insertTime() {
+    return insertTime;
+  }
+
+  /**
+   * Returns the time that this operation was started by the service. In milliseconds since epoch.
+   */
+  public Long startTime() {
+    return startTime;
+  }
+
+  /**
+   * Returns the time that this operation was completed. In milliseconds since epoch.
+   */
+  public Long endTime() {
+    return endTime;
+  }
+
+  /**
+   * Returns the errors encountered while processing this operation, if any. Returns {@code null} if
```

Comment:
Did we decide how to proceed with respect to #664? Should `errors` return an empty list even if the user is not selecting the `OperationField.ERROR` field? This returns `null` (as we do in most of the rest of the library) to avoid taking actions for #664 in PRs that are not directly related.